lib/std/net.zig

```diff
@@ -214,8 +218,28 @@ pub const IpAddress = extern union {
         };
     }
 
+    pub fn initUnix(path: []const u8) !Address {
+        switch (builtin.os) {
+            .linux, .macosx, .freebsd, .netbsd => {},
+            else => return error.UnsupportedOS,
+        }
+
+        var sock_addr = os.sockaddr_un{
+            .family = os.AF_UNIX,
+            .path = undefined,
+        };
+
+        // this enables us to have the proper length of the socket in getOsSockLen
+        mem.set(u8, &sock_addr.path, 0);
+
+        if (path.len > sock_addr.path.len) return error.NameTooLong;
+        mem.copy(u8, &sock_addr.path, path);
+
+        return Address{ .un = sock_addr };
+    }
+
     /// Returns the port in native endian.
-    pub fn getPort(self: IpAddress) u16 {
+    pub fn getPort(self: Address) u16 {
```

Comment:
Can you add doc comments here and for `setPort` to make note that these functions assert the address family is ip6 or ip4? Callers should know that this will invoke unreachable if the address is a unix socket.